From 49bc7e6ae9f8e062a8640f00716594bc408b2e61 Mon Sep 17 00:00:00 2001
From: Jonathan Scruggs <j.scruggs@gmail.com>
Date: Tue, 5 Sep 2017 11:40:06 +0100
Subject: [PATCH] Change hardcoded paths to GNUInstallDirs variables.

According to standards, 64 bit files need to be installed to lib64
http://www.pathname.com/fhs/pub/fhs-2.3.html#LIB64

GNUInstallDirs enforces the standards on all systems, Linux,
Windows, MacOS, and BSD. These variables can be overridden.

Fix bug of not being able to set Python install dir.

Signed-off by: Jonathan Scruggs <j.scruggs@gmail.com>
---
 CMakeLists.txt                 |  2 ++
 cmake/FindBlosc.cmake          |  6 +++---
 cmake/FindCPPUnit.cmake        |  7 ++++---
 cmake/FindGLEW.cmake           | 33 +++++++++++++++++----------------
 cmake/FindGLFW.cmake           | 19 ++++++++++---------
 cmake/FindGLFW3.cmake          |  9 +++++----
 cmake/FindILMBase.cmake        | 12 ++++++------
 cmake/FindMaya.cmake           |  8 ++++----
 cmake/FindOpenEXR.cmake        | 12 ++++++------
 cmake/FindOpenVDB.cmake        |  6 +++---
 cmake/FindTBB.cmake            | 10 +++++-----
 openvdb/CMakeLists.txt         | 16 ++++++++--------
 openvdb/python/CMakeLists.txt  | 10 ++++++----
 openvdb_houdini/CMakeLists.txt |  2 +-
 14 files changed, 80 insertions(+), 72 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 712fb1d..1d02e9e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,6 +54,8 @@ ENDIF ()
 
 LIST(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
+INCLUDE ( GNUInstallDirs )
+
 SET ( OPENVDB_TOP_LEVEL_DIR ${PROJECT_SOURCE_DIR} CACHE PATH "OpenVDB Top Level Directory to share among sub projects" )
 
 INCLUDE_DIRECTORIES ( ${OPENVDB_TOP_LEVEL_DIR} ) # To make sure we pick up headers from this version of OpenVDB we are building
diff --git a/cmake/FindBlosc.cmake b/cmake/FindBlosc.cmake
index 2d9d9d3..5efcea4 100644
--- a/cmake/FindBlosc.cmake
+++ b/cmake/FindBlosc.cmake
@@ -37,7 +37,7 @@
 
 FIND_PACKAGE ( PackageHandleStandardArgs )
 
-FIND_PATH( BLOSC_LOCATION include/blosc.h
+FIND_PATH( BLOSC_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/blosc.h
   "$ENV{BLOSC_ROOT}"
   NO_DEFAULT_PATH
   NO_SYSTEM_ENVIRONMENT_PATH
@@ -49,7 +49,7 @@ FIND_PACKAGE_HANDLE_STANDARD_ARGS ( Blosc
 
 IF ( BLOSC_FOUND )
 
-  SET ( BLOSC_LIBRARYDIR ${BLOSC_LOCATION}/lib
+  SET ( BLOSC_LIBRARYDIR ${BLOSC_LOCATION}/${CMAKE_INSTALL_LIBDIR}
     CACHE STRING "Blosc library directories")
 
   SET ( _blosc_library_name "blosc" )
@@ -75,6 +75,6 @@ IF ( BLOSC_FOUND )
     SET( CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES_BACKUP} )
   ENDIF()
 
-  SET( BLOSC_INCLUDE_DIR "${BLOSC_LOCATION}/include" CACHE STRING "Blosc include directory" )
+  SET( BLOSC_INCLUDE_DIR "${BLOSC_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}" CACHE STRING "Blosc include directory" )
 
 ENDIF ( BLOSC_FOUND )
diff --git a/cmake/FindCPPUnit.cmake b/cmake/FindCPPUnit.cmake
index 4ed56e9..afa7b63 100644
--- a/cmake/FindCPPUnit.cmake
+++ b/cmake/FindCPPUnit.cmake
@@ -38,7 +38,7 @@
 
 FIND_PACKAGE ( PackageHandleStandardArgs )
 
-FIND_PATH( CPPUNIT_LOCATION include/cppunit/Test.h
+FIND_PATH( CPPUNIT_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/cppunit/Test.h
   "$ENV{CPPUNIT_ROOT}"
   NO_DEFAULT_PATH
   NO_CMAKE_ENVIRONMENT_PATH
@@ -53,13 +53,14 @@ FIND_PACKAGE_HANDLE_STANDARD_ARGS ( CPPUnit
 
 IF ( CPPUNIT_FOUND )
   
-  SET( CPPUNIT_INCLUDE_DIR "${CPPUNIT_LOCATION}/include" CACHE STRING "CPPUNIT include directory")
+  SET( CPPUNIT_INCLUDE_DIR "${CPPUNIT_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}"
+       CACHE STRING "CPPUNIT include directory")
   IF (CPPUnit_USE_STATIC_LIBS)
 	SET( CPPUNIT_LIBRARY_NAME libcppunit.a)
   ELSE ()
 	SET( CPPUNIT_LIBRARY_NAME cppunit)
   ENDIF ()
   FIND_LIBRARY ( CPPUnit_cppunit_LIBRARY ${CPPUNIT_LIBRARY_NAME}
-	PATHS ${CPPUNIT_LOCATION}/lib )
+	PATHS ${CPPUNIT_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
   
 ENDIF ( CPPUNIT_FOUND )
diff --git a/cmake/FindGLEW.cmake b/cmake/FindGLEW.cmake
index 0b72457..730280d 100644
--- a/cmake/FindGLEW.cmake
+++ b/cmake/FindGLEW.cmake
@@ -39,7 +39,7 @@
 
 FIND_PACKAGE ( PackageHandleStandardArgs )
 
-FIND_PATH( GLEW_LOCATION include/GL/glew.h
+FIND_PATH( GLEW_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/GL/glew.h
   "$ENV{GLEW_ROOT}"
   NO_DEFAULT_PATH
   NO_SYSTEM_ENVIRONMENT_PATH
@@ -51,17 +51,18 @@ FIND_PACKAGE_HANDLE_STANDARD_ARGS ( GLEW
 
 IF ( GLEW_LOCATION )
 
-  SET( GLEW_INCLUDE_DIR "${GLEW_LOCATION}/include" CACHE STRING "GLEW include path")
+  SET( GLEW_INCLUDE_DIR "${GLEW_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}"
+       CACHE STRING "GLEW include path")
 
   SET ( ORIGINAL_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
   IF (GLEW_USE_STATIC_LIBS)
 	IF (APPLE)
       SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
-      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW PATHS ${GLEW_LOCATION}/lib
+      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
-      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEWmx PATHS ${GLEW_LOCATION}/lib
+      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEWmx PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
@@ -70,15 +71,15 @@ IF ( GLEW_LOCATION )
 	ELSEIF (WIN32)
       # Link library
       SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW32S PATHS ${GLEW_LOCATION}/lib )
-      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEW32MXS PATHS ${GLEW_LOCATION}/lib )
+      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW32S PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
+      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEW32MXS PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
 	ELSE (APPLE)
       SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
-      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW PATHS ${GLEW_LOCATION}/lib
+      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
-      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEWmx PATHS ${GLEW_LOCATION}/lib
+      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEWmx PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
@@ -88,30 +89,30 @@ IF ( GLEW_LOCATION )
   ELSE ()
 	IF (APPLE)
       SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dylib")
-      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW PATHS ${GLEW_LOCATION}/lib )
-      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEWmx PATHS ${GLEW_LOCATION}/lib )
+      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
+      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEWmx PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
 	ELSEIF (WIN32)
       # Link library
       SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW32 PATHS ${GLEW_LOCATION}/lib )
-      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEW32mx PATHS ${GLEW_LOCATION}/lib )
+      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW32 PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
+      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEW32mx PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
       # Load library
       SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
-      FIND_LIBRARY ( GLEW_DLL_PATH GLEW32 PATHS ${GLEW_LOCATION}/bin
+      FIND_LIBRARY ( GLEW_DLL_PATH GLEW32 PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_BINDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
-      FIND_LIBRARY ( GLEWmx_DLL_PATH GLEW32mx PATHS ${GLEW_LOCATION}/bin
+      FIND_LIBRARY ( GLEWmx_DLL_PATH GLEW32mx PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_BINDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
 	ELSE (APPLE)
 	  # Unices
-      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW PATHS ${GLEW_LOCATION}/lib
+      FIND_LIBRARY ( GLEW_LIBRARY_PATH GLEW PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
-      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEWmx PATHS ${GLEW_LOCATION}/lib
+      FIND_LIBRARY ( GLEWmx_LIBRARY_PATH GLEWmx PATHS ${GLEW_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
diff --git a/cmake/FindGLFW.cmake b/cmake/FindGLFW.cmake
index 1e7cc08..f58b9ad 100644
--- a/cmake/FindGLFW.cmake
+++ b/cmake/FindGLFW.cmake
@@ -38,19 +38,20 @@
 
 SET ( GLFW_FOUND FALSE )
 
-FIND_PATH( GLFW_LOCATION include/GL/glfw.h
+FIND_PATH( GLFW_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/GL/glfw.h
   "$ENV{GLFW_ROOT}"
   NO_DEFAULT_PATH
   NO_SYSTEM_ENVIRONMENT_PATH
   )
 
-SET( GLFW_INCLUDE_DIR "${GLFW_LOCATION}/include" CACHE STRING "GLFW include path")
+SET( GLFW_INCLUDE_DIR "${GLFW_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}"
+     CACHE STRING "GLFW include path")
 
 SET ( ORIGINAL_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
 IF (GLFW_USE_STATIC_LIBS)
   IF (APPLE)
     SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
-    FIND_LIBRARY ( GLFW_LIBRARY_PATH GLFW PATHS ${GLFW_LOCATION}/lib
+    FIND_LIBRARY ( GLFW_LIBRARY_PATH GLFW PATHS ${GLFW_LOCATION}/${CMAKE_INSTALL_LIBDIR}
       NO_DEFAULT_PATH
       NO_SYSTEM_ENVIRONMENT_PATH
 	  )
@@ -59,10 +60,10 @@ IF (GLFW_USE_STATIC_LIBS)
   ELSEIF (WIN32)
     # Link library
     SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-    FIND_LIBRARY ( GLFW_LIBRARY_PATH GLFW32S PATHS ${GLFW_LOCATION}/lib )
+    FIND_LIBRARY ( GLFW_LIBRARY_PATH GLFW32S PATHS ${GLFW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
   ELSE (APPLE)
     SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
-    FIND_LIBRARY ( GLFW_LIBRARY_PATH glfw PATHS ${GLFW_LOCATION}/lib ${GLFW_LOCATION}/lib64
+    FIND_LIBRARY ( GLFW_LIBRARY_PATH glfw PATHS ${GLFW_LOCATION}/${CMAKE_INSTALL_LIBDIR} ${GLFW_LOCATION}/lib64
       NO_DEFAULT_PATH
       NO_SYSTEM_ENVIRONMENT_PATH
 	  )
@@ -72,20 +73,20 @@ IF (GLFW_USE_STATIC_LIBS)
 ELSE ()
   IF (APPLE)
     SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dylib")
-    FIND_LIBRARY ( GLFW_LIBRARY_PATH GLFW PATHS ${GLFW_LOCATION}/lib )
+    FIND_LIBRARY ( GLFW_LIBRARY_PATH GLFW PATHS ${GLFW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
   ELSEIF (WIN32)
     # Link library
     SET(CMAKE_FIND_LIBRARY_SUFFIXES ".lib")
-    FIND_LIBRARY ( GLFW_LIBRARY_PATH GLFW32 PATHS ${GLFW_LOCATION}/lib )
+    FIND_LIBRARY ( GLFW_LIBRARY_PATH GLFW32 PATHS ${GLFW_LOCATION}/${CMAKE_INSTALL_LIBDIR} )
     # Load library
     SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
-    FIND_LIBRARY ( GLFW_DLL_PATH GLFW32 PATHS ${GLFW_LOCATION}/bin
+    FIND_LIBRARY ( GLFW_DLL_PATH GLFW32 PATHS ${GLFW_LOCATION}/${CMAKE_INSTALL_BINDIR}
       NO_DEFAULT_PATH
       NO_SYSTEM_ENVIRONMENT_PATH
       )
   ELSE (APPLE)
 	# Unices
-    FIND_LIBRARY ( GLFW_LIBRARY_PATH glfw PATHS ${GLFW_LOCATION}/lib ${GLFW_LOCATION}/lib64
+    FIND_LIBRARY ( GLFW_LIBRARY_PATH glfw PATHS ${GLFW_LOCATION}/${CMAKE_INSTALL_LIBDIR}
       NO_DEFAULT_PATH
       NO_SYSTEM_ENVIRONMENT_PATH
 	  )
diff --git a/cmake/FindGLFW3.cmake b/cmake/FindGLFW3.cmake
index 88e3a76..f9d9df7 100644
--- a/cmake/FindGLFW3.cmake
+++ b/cmake/FindGLFW3.cmake
@@ -39,7 +39,7 @@
 
 FIND_PACKAGE ( PackageHandleStandardArgs )
 
-FIND_PATH( GLFW3_LOCATION include/GLFW/glfw3.h
+FIND_PATH( GLFW3_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/GLFW/glfw3.h
   "$ENV{GLFW3_ROOT}"
   NO_DEFAULT_PATH
   NO_SYSTEM_ENVIRONMENT_PATH
@@ -50,9 +50,10 @@ FIND_PACKAGE_HANDLE_STANDARD_ARGS ( GLFW3
   )
 
 IF (GLFW3_FOUND)
-  SET( GLFW3_INCLUDE_DIR "${GLFW3_LOCATION}/include" CACHE STRING "GLFW3 include path")
+  SET( GLFW3_INCLUDE_DIR "${GLFW3_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}"
+       CACHE STRING "GLFW3 include path")
   IF (GLFW3_USE_STATIC_LIBS)
-    FIND_LIBRARY ( GLFW3_glfw_LIBRARY  glfw3  ${GLFW3_LOCATION}/lib
+    FIND_LIBRARY ( GLFW3_glfw_LIBRARY  glfw3  ${GLFW3_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 	  NO_DEFAULT_PATH
 	  NO_CMAKE_ENVIRONMENT_PATH
 	  NO_CMAKE_PATH
@@ -60,7 +61,7 @@ IF (GLFW3_FOUND)
 	  NO_CMAKE_SYSTEM_PATH
 	  )
   ELSE (GLFW3_USE_STATIC_LIBS)
-    FIND_LIBRARY ( GLFW3_glfw_LIBRARY  glfw  ${GLFW3_LOCATION}/lib
+    FIND_LIBRARY ( GLFW3_glfw_LIBRARY  glfw  ${GLFW3_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 	  NO_DEFAULT_PATH
 	  NO_CMAKE_ENVIRONMENT_PATH
 	  NO_CMAKE_PATH
diff --git a/cmake/FindILMBase.cmake b/cmake/FindILMBase.cmake
index 664affd..b803b87 100644
--- a/cmake/FindILMBase.cmake
+++ b/cmake/FindILMBase.cmake
@@ -43,7 +43,7 @@
 
 FIND_PACKAGE ( PackageHandleStandardArgs )
 
-FIND_PATH ( ILMBASE_LOCATION include/OpenEXR/IlmBaseConfig.h
+FIND_PATH ( ILMBASE_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/IlmBaseConfig.h
   "$ENV{ILMBASE_ROOT}"
   NO_DEFAULT_PATH
   NO_SYSTEM_ENVIRONMENT_PATH
@@ -57,11 +57,11 @@ OPTION ( ILMBASE_NAMESPACE_VERSIONING "Namespace versioning of libraries" ON )
 
 IF ( ILMBASE_FOUND )
 
-  FILE ( STRINGS "${ILMBASE_LOCATION}/include/OpenEXR/IlmBaseConfig.h" _ilmbase_version_major_string REGEX "#define ILMBASE_VERSION_MAJOR ")
+  FILE ( STRINGS "${ILMBASE_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/IlmBaseConfig.h" _ilmbase_version_major_string REGEX "#define ILMBASE_VERSION_MAJOR ")
   STRING ( REGEX REPLACE "#define ILMBASE_VERSION_MAJOR" "" _ilmbase_version_major_unstrip "${_ilmbase_version_major_string}")
   STRING ( STRIP "${_ilmbase_version_major_unstrip}" ILMBASE_VERSION_MAJOR )
 
-  FILE ( STRINGS "${ILMBASE_LOCATION}/include/OpenEXR/IlmBaseConfig.h" _ilmbase_version_minor_string REGEX "#define ILMBASE_VERSION_MINOR ")
+  FILE ( STRINGS "${ILMBASE_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/IlmBaseConfig.h" _ilmbase_version_minor_string REGEX "#define ILMBASE_VERSION_MINOR ")
   STRING ( REGEX REPLACE "#define ILMBASE_VERSION_MINOR" "" _ilmbase_version_minor_unstrip "${_ilmbase_version_minor_string}")
   STRING ( STRIP "${_ilmbase_version_minor_unstrip}" ILMBASE_VERSION_MINOR )
 
@@ -78,10 +78,10 @@ IF ( ILMBASE_FOUND )
   ENDIF ( ILMBASE_NAMESPACE_VERSIONING )
 
   SET ( ILMBASE_INCLUDE_DIRS
-    ${ILMBASE_LOCATION}/include
-    ${ILMBASE_LOCATION}/include/OpenEXR
+    ${ILMBASE_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}
+    ${ILMBASE_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR
     CACHE STRING "ILMBase include directories")
-  SET ( ILMBASE_LIBRARYDIR ${ILMBASE_LOCATION}/lib
+  SET ( ILMBASE_LIBRARYDIR ${ILMBASE_LOCATION}/${CMAKE_INSTALL_LIBDIR}
     CACHE STRING "ILMBase library directories")
   SET ( ILMBASE_FOUND TRUE )
   
diff --git a/cmake/FindMaya.cmake b/cmake/FindMaya.cmake
index 7e48f80..a847ca0 100644
--- a/cmake/FindMaya.cmake
+++ b/cmake/FindMaya.cmake
@@ -40,7 +40,7 @@
 
 FIND_PACKAGE ( PackageHandleStandardArgs )
 
-FIND_PATH( MAYA_LOCATION include/maya/MTypes.h
+FIND_PATH( MAYA_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/maya/MTypes.h
   "$ENV{MAYA_LOCATION}"
   NO_DEFAULT_PATH
   NO_CMAKE_ENVIRONMENT_PATH
@@ -55,7 +55,7 @@ FIND_PACKAGE_HANDLE_STANDARD_ARGS ( Maya
 
 IF ( MAYA_FOUND )
 
-  SET( MAYA_INCLUDE_DIR       "${MAYA_LOCATION}/include" CACHE STRING "Maya include directory")
+  SET( MAYA_INCLUDE_DIR       "${MAYA_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}" CACHE STRING "Maya include directory")
 
   # Determine Maya version, including point releases
   # Currently only works for Maya 2016 and onwards so
@@ -88,7 +88,7 @@ IF ( MAYA_FOUND )
   
   FOREACH ( lib_component ${MAYA_LIBRARY_COMPONENTS} )
     FIND_LIBRARY ( MAYA_${lib_component}_LIBRARY  ${lib_component}
-      PATHS ${MAYA_LOCATION}/lib
+      PATHS ${MAYA_LOCATION}/${CMAKE_INSTALL_LIDIR}
 	  NO_DEFAULT_PATH
       NO_SYSTEM_ENVIRONMENT_PATH
       )
@@ -108,7 +108,7 @@ IF ( MAYA_FOUND )
 	
 	FOREACH ( lib_component ${MAYA_BOOST_LIBRARY_COMPONENTS} )
       FIND_LIBRARY ( MAYA_${lib_component}_LIBRARY  ${lib_component}-${MAYA_BOOST_VERSION}
-		PATHS ${MAYA_LOCATION}/lib
+		PATHS ${MAYA_LOCATION}/${CMAKE_INSTALL_LIBDIR}
 		NO_DEFAULT_PATH
 		NO_SYSTEM_ENVIRONMENT_PATH
 		)
diff --git a/cmake/FindOpenEXR.cmake b/cmake/FindOpenEXR.cmake
index 6d69c03..1fa4875 100644
--- a/cmake/FindOpenEXR.cmake
+++ b/cmake/FindOpenEXR.cmake
@@ -38,7 +38,7 @@
 
 FIND_PACKAGE ( PackageHandleStandardArgs )
 
-FIND_PATH ( OPENEXR_LOCATION include/OpenEXR/OpenEXRConfig.h
+FIND_PATH ( OPENEXR_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/OpenEXRConfig.h
   ENV OPENEXR_ROOT
   NO_DEFAULT_PATH
   NO_SYSTEM_ENVIRONMENT_PATH
@@ -52,11 +52,11 @@ OPTION ( OPENEXR_NAMESPACE_VERSIONING "Namespace versioning of libraries" ON )
 
 IF ( OPENEXR_FOUND )
 
-  FILE ( STRINGS "${OPENEXR_LOCATION}/include/OpenEXR/OpenEXRConfig.h" _openexr_version_major_string REGEX "#define OPENEXR_VERSION_MAJOR ")
+  FILE ( STRINGS "${OPENEXR_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/OpenEXRConfig.h" _openexr_version_major_string REGEX "#define OPENEXR_VERSION_MAJOR ")
   STRING ( REGEX REPLACE "#define OPENEXR_VERSION_MAJOR" "" _openexr_version_major_unstrip "${_openexr_version_major_string}")
   STRING ( STRIP "${_openexr_version_major_unstrip}" OPENEXR_VERSION_MAJOR )
 
-  FILE ( STRINGS "${OPENEXR_LOCATION}/include/OpenEXR/OpenEXRConfig.h" _openexr_version_minor_string REGEX "#define OPENEXR_VERSION_MINOR ")
+  FILE ( STRINGS "${OPENEXR_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR/OpenEXRConfig.h" _openexr_version_minor_string REGEX "#define OPENEXR_VERSION_MINOR ")
   STRING ( REGEX REPLACE "#define OPENEXR_VERSION_MINOR" "" _openexr_version_minor_unstrip "${_openexr_version_minor_string}")
   STRING ( STRIP "${_openexr_version_minor_unstrip}" OPENEXR_VERSION_MINOR )
   
@@ -69,10 +69,10 @@ IF ( OPENEXR_FOUND )
   ENDIF ( OPENEXR_NAMESPACE_VERSIONING )
 	
   SET ( OPENEXR_INCLUDE_DIRS
-    ${OPENEXR_LOCATION}/include
-    ${OPENEXR_LOCATION}/include/OpenEXR
+    ${OPENEXR_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}
+    ${OPENEXR_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}/OpenEXR
     CACHE STRING "Openexr include directories")
-  SET ( OPENEXR_LIBRARYDIR ${OPENEXR_LOCATION}/lib
+  SET ( OPENEXR_LIBRARYDIR ${OPENEXR_LOCATION}/${CMAKE_INSTALL_LIBDIR}
     CACHE STRING "Openexr library directories")
   SET ( OPENEXR_FOUND TRUE )
 
diff --git a/cmake/FindOpenVDB.cmake b/cmake/FindOpenVDB.cmake
index 1d34767..ce09007 100644
--- a/cmake/FindOpenVDB.cmake
+++ b/cmake/FindOpenVDB.cmake
@@ -45,7 +45,7 @@
 
 FIND_PACKAGE( PackageHandleStandardArgs )
 
-FIND_PATH( OPENVDB_LOCATION include/openvdb/version.h 
+FIND_PATH( OPENVDB_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/openvdb/version.h 
   "ENV{OPENVDB_ROOT}"
   NO_DEFAULT_PATH
   NO_SYSTEM_ENVIRONMENT_PATH
@@ -56,10 +56,10 @@ FIND_PACKAGE_HANDLE_STANDARD_ARGS( OpenVDB
   )
 
 IF( OpenVDB_FOUND )
-  SET( OpenVDB_INCLUDE_DIR ${OPENVDB_LOCATION}/include
+  SET( OpenVDB_INCLUDE_DIR ${OPENVDB_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}
     CACHE PATH "OpenVDB include directory")
 
-  SET( OpenVDB_LIBRARY_DIR ${OPENVDB_LOCATION}/lib
+  SET( OpenVDB_LIBRARY_DIR ${OPENVDB_LOCATION}/${CMAKE_INSTALL_LIBDIR}
     CACHE PATH "OpenVDB library directory" )
   
   FIND_LIBRARY( OpenVDB_OPENVDB_LIBRARY openvdb
diff --git a/cmake/FindTBB.cmake b/cmake/FindTBB.cmake
index 3fe6446..3615940 100644
--- a/cmake/FindTBB.cmake
+++ b/cmake/FindTBB.cmake
@@ -42,7 +42,7 @@ FIND_PACKAGE ( PackageHandleStandardArgs )
 
 # SET ( TBB_FOUND FALSE )
 
-FIND_PATH( TBB_LOCATION include/tbb/tbb.h
+FIND_PATH( TBB_LOCATION ${CMAKE_INSTALL_INCLUDEDIR}/tbb/tbb.h
   "$ENV{TBB_ROOT}"
   NO_DEFAULT_PATH
   NO_CMAKE_ENVIRONMENT_PATH
@@ -57,13 +57,13 @@ FIND_PACKAGE_HANDLE_STANDARD_ARGS ( TBB
 
 IF ( TBB_FOUND )
 
-  SET( TBB_INCLUDE_DIR "${TBB_LOCATION}/include" CACHE STRING "TBB include directory")
+  SET( TBB_INCLUDE_DIR "${TBB_LOCATION}/${CMAKE_INSTALL_INCLUDEDIR}" CACHE STRING "TBB include directory")
 
   IF (APPLE)
 	IF (TBB_FOR_CLANG)
-      SET ( TBB_LIBRARYDIR ${TBB_LOCATION}/lib/libc++ CACHE STRING "TBB library directory")
+      SET ( TBB_LIBRARYDIR ${TBB_LOCATION}/${CMAKE_INSTALL_LIBDIR}/libc++ CACHE STRING "TBB library directory")
 	ELSE ()
-      SET ( TBB_LIBRARYDIR ${TBB_LOCATION}/lib CACHE STRING "TBB library directory")
+      SET ( TBB_LIBRARYDIR ${TBB_LOCATION}/${CMAKE_INSTALL_LIBDIR} CACHE STRING "TBB library directory")
 	ENDIF ()
 	SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dylib")
 	FIND_LIBRARY ( TBB_LIBRARY_PATH tbb PATHS ${TBB_LIBRARYDIR} )
@@ -86,7 +86,7 @@ IF ( TBB_FOUND )
 	LIST ( APPEND TBB_LIBRARIES_LIST ${TBB_LIBRARY_PATH} ${TBBmx_LIBRARY_PATH} )
   ELSE (APPLE)
 	# MESSAGE ( "CMAKE_COMPILER_IS_GNUCXX = ${CMAKE_COMPILER_IS_GNUCXX}")
-    SET ( TBB_LIBRARYDIR ${TBB_LOCATION}/lib CACHE STRING "TBB library directory")
+    SET ( TBB_LIBRARYDIR ${TBB_LOCATION}/${CMAKE_INSTALL_LIBDIR} CACHE STRING "TBB library directory")
 	IF (${CMAKE_COMPILER_IS_GNUCXX})
 	  IF ( TBB_MATCH_COMPILER_VERSION )
 		STRING(REGEX MATCHALL "[0-9]+" GCC_VERSION_COMPONENTS ${CMAKE_CXX_COMPILER_VERSION})
diff --git a/openvdb/CMakeLists.txt b/openvdb/CMakeLists.txt
index ca91234..790f627 100644
--- a/openvdb/CMakeLists.txt
+++ b/openvdb/CMakeLists.txt
@@ -401,7 +401,7 @@ IF (OPENVDB_BUILD_DOCS)
     )
 
   INSTALL ( CODE "EXECUTE_PROCESS(COMMAND \"${CMAKE_COMMAND}\" --build  \"${PROJECT_BINARY_DIR}\" --target doc)")
-  INSTALL ( DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/doc/html DESTINATION docs)
+  INSTALL ( DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/doc/html DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
 ENDIF ()
 
@@ -410,37 +410,37 @@ IF ( NOT WIN32 )
   INSTALL ( TARGETS
     vdb_view
     DESTINATION
-    bin
+    ${CMAKE_INSTALL_BINDIR}
     )
 ENDIF ()
 INSTALL ( TARGETS
   vdb_print
   vdb_render
   DESTINATION
-  bin
+  ${CMAKE_INSTALL_BINDIR}
   )
 
 INSTALL ( TARGETS
   openvdb_static
   DESTINATION
-  lib
+  ${CMAKE_INSTALL_LIBDIR}
   )
 
 IF (WIN32)
   INSTALL ( TARGETS
 	openvdb_shared
-	RUNTIME DESTINATION bin COMPONENT Runtime
-	ARCHIVE DESTINATION lib COMPONENT Runtime
+	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT Runtime
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Runtime
 	)
 ELSE()
   INSTALL ( TARGETS
 	openvdb_shared
 	DESTINATION
-	lib
+	${CMAKE_INSTALL_LIBDIR}
 	)
 ENDIF ()
 
-INSTALL ( DIRECTORY ${PROJECT_SOURCE_DIR} DESTINATION include
+INSTALL ( DIRECTORY ${PROJECT_SOURCE_DIR} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
   FILES_MATCHING PATTERN "*.h"
   PATTERN "python" EXCLUDE
   PATTERN "cmd" EXCLUDE
diff --git a/openvdb/python/CMakeLists.txt b/openvdb/python/CMakeLists.txt
index f7b7f57..149b720 100644
--- a/openvdb/python/CMakeLists.txt
+++ b/openvdb/python/CMakeLists.txt
@@ -54,10 +54,12 @@ SET ( OPENVDB_PYTHON_MODULE_SOURCE_FILES
   pyVec3Grid.cc
   )
 
-SET ( PYOPENVDB_INSTALL_DIRECTORY
-  lib/python${OPENVDB_PYTHON_VERSION_MAJOR}.${OPENVDB_PYTHON_VERSION_MINOR}
-  CACHE STRING "The directory to install the pyopenvdb.so module."
-  )
+IF ( NOT DEFINED PYOPENVDB_INSTALL_DIRECTORY)
+  SET ( PYOPENVDB_INSTALL_DIRECTORY
+    ${CMAKE_INSTALL_LIBDIR}/python${OPENVDB_PYTHON_VERSION_MAJOR}.${OPENVDB_PYTHON_VERSION_MINOR}/site-packages
+    CACHE STRING "The directory to install the pyopenvdb.so module."
+    )
+ENDIF ()
 
 SET_SOURCE_FILES_PROPERTIES ( ${OPENVDB_PYTHON_MODULE_SOURCE_FILES}
   PROPERTIES
diff --git a/openvdb_houdini/CMakeLists.txt b/openvdb_houdini/CMakeLists.txt
index d336128..3c128bf 100644
--- a/openvdb_houdini/CMakeLists.txt
+++ b/openvdb_houdini/CMakeLists.txt
@@ -221,7 +221,7 @@ IF( OPENVDB_HOUDINI_INSTALL_LIBRARY )
   INSTALL ( TARGETS
     openvdb_houdini
     DESTINATION
-    ${OPENVDB_HFS_INSTALL_BASE_DIR}/lib
+    ${OPENVDB_HFS_INSTALL_BASE_DIR}/${CMAKE_INSTALL_LIBDIR}
     )
 
 ELSE ()
-- 
2.14.1

